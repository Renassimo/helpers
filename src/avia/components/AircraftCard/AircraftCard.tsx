import { showTimePassed, showWhen } from '@/common/utils/dayjs';

import { Avia } from '@/avia/types/avia';

import AviaCard from '@/avia/components/AviaCard';

const AircraftCard = ({
  data,
  onClick,
  chosen = false,
}: {
  data: Avia.AircraftData;
  onClick: () => void;
  chosen?: boolean;
}) => {
  const { attributes } = data;
  const {
    registration,
    serial,
    airlineName,
    typeName,
    productionLine,
    modelCode,
    model,
    photoUrl,
    firstFlightDate,
    rolloutDate,
    deliveryDate,
  } = attributes;

  const date = firstFlightDate ?? rolloutDate ?? deliveryDate;

  const additionalContent: (string | null)[] = [
    airlineName,
    typeName || productionLine,
    modelCode || model,
    serial && `CN: ${serial}`,
    date && `From ${showWhen(date, false)} (${showTimePassed(date)})`,
  ];
  const link =
    registration && `https://www.planespotters.net/search?q=${registration}`;

  return (
    <AviaCard
      onClick={onClick}
      chosen={chosen}
      imageUrl={photoUrl ?? '/images/plane.png'}
      imageAlt={registration}
      link={link}
      title={registration}
      additionalContent={additionalContent}
    />
  );
};

export default AircraftCard;
