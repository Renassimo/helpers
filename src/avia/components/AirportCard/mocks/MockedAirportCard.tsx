const MockedAirportCard = jest.fn(({ title, chosen }) => (
  <div>
    MockedAirportCard - {title} - {chosen && 'chosen'}
  </div>
));

export default MockedAirportCard;
