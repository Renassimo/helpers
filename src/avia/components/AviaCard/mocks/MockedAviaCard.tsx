const MockedAviaCard = jest.fn(
  ({ additionalContent, chosen = false, imageUrl, imageAlt, title }) => (
    <div>
      MockedAviaCard - {title} -
      {additionalContent.map(
        (content: string | null, index: number) =>
          content && <div key={content + index}>{content}</div>
      )}{' '}
      - {chosen && 'chosen'} - {imageUrl} - {imageAlt}
    </div>
  )
);

export default MockedAviaCard;
