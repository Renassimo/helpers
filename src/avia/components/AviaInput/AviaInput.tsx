import { KeyboardEvent, ReactNode } from 'react';

import FormControl from '@mui/material/FormControl';
import Input from '@mui/material/Input';
import InputAdornment from '@mui/material/InputAdornment';
import InputLabel from '@mui/material/InputLabel';

import IconButton from '@mui/material/IconButton';
import CloseIcon from '@mui/icons-material/Close';

const AviaInput = ({
  title,
  placeholder,
  label,
  value,
  setValue,
  disabled,
  onKeyDown,
  adornments,
}: {
  title: string;
  placeholder?: string;
  label?: string;
  value: string;
  setValue: (value: string) => void;
  disabled: boolean;
  onKeyDown?: (event: KeyboardEvent) => void;
  adornments: ReactNode;
}) => {
  return (
    <FormControl fullWidth variant="standard">
      {label && <InputLabel>{label}</InputLabel>}
      <Input
        value={value}
        onChange={(event) => setValue(event.target.value)}
        size="small"
        disabled={disabled}
        placeholder={placeholder ?? title}
        aria-label={title}
        fullWidth
        onKeyDown={onKeyDown}
        endAdornment={
          <InputAdornment position="end">
            <>
              {adornments}
              {value && (
                <IconButton
                  size="small"
                  aria-label="Clear"
                  onClick={() => setValue('')}
                  disabled={disabled}
                  edge="end"
                >
                  {<CloseIcon />}
                </IconButton>
              )}
            </>
          </InputAdornment>
        }
      />
    </FormControl>
  );
};

export default AviaInput;
