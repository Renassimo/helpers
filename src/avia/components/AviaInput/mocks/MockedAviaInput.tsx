const MockedAviaInput = jest.fn(
  ({ title, placeholder, label, value, disabled, adornments }) => (
    <div>
      MockedAviaInput - {title} - {placeholder} - {label} - {value} -{' '}
      {disabled && 'disabled'} - {adornments}
    </div>
  )
);

export default MockedAviaInput;
