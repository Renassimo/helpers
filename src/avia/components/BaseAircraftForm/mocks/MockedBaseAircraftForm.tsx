const MockedBaseAircraftForm = jest.fn(({ registration }) => (
  <div>MockedBaseAircraftForm - {registration}</div>
));

export default MockedBaseAircraftForm;
