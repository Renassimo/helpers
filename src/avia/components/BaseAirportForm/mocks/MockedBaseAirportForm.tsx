const MockedBaseAirportForm = jest.fn(({ title }) => (
  <div>MockedBaseAirportForm - {title}</div>
));

export default MockedBaseAirportForm;
