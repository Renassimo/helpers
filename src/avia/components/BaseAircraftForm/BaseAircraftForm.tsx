import { KeyboardEvent, useEffect } from 'react';

import { Avia } from '@/avia/types/avia';

import Box from '@mui/material/Box';
import IconButton from '@mui/material/IconButton';
import SearchIcon from '@mui/icons-material/Search';

import useStateValue from '@/common/hooks/useStateValue';

import AviaInput from '@/avia/components/AviaInput';
import AircraftCard from '@/avia/components/AircraftCard';

const BaseAircraftForm = ({
  registration = null,
  aircraftsResult: {
    aircrafts,
    chosenAircraft,
    retreiveAircrafts,
    chooseAircraft,
    clearChosenAircraft,
    loading,
  },
}: {
  registration?: string | null;
  aircraftsResult: Avia.AircraftsResult;
}) => {
  const [aircraftValue, setAircraftValue] = useStateValue('');

  const onSubmit = async () => {
    await retreiveAircrafts(aircraftValue);
  };

  const lengthOk = aircraftValue.length > 2;

  const onEnter = (event: KeyboardEvent) => {
    if (event.key === 'Enter' && lengthOk) onSubmit();
  };

  useEffect(() => {
    if (!aircraftValue && registration) setAircraftValue(registration);
  }, [registration]);

  return (
    <Box>
      {!chosenAircraft && (
        <Box mt={3}>
          <AviaInput
            title="Aircraft registration"
            value={aircraftValue}
            setValue={setAircraftValue}
            disabled={loading}
            onKeyDown={onEnter}
            adornments={
              <IconButton
                size="small"
                onClick={onSubmit}
                disabled={loading || !lengthOk}
                aria-label="Search"
              >
                <SearchIcon />
              </IconButton>
            }
          />
        </Box>
      )}
      {aircrafts && !chosenAircraft && (
        <Box display="flex" overflow="auto">
          {aircrafts.map((aircraft, index) => (
            <Box
              key={aircraft.id + index}
              mr={1}
              ml={index === 0 ? 1 : undefined}
            >
              <AircraftCard
                data={aircraft}
                onClick={() => chooseAircraft(aircraft.id)}
              />
            </Box>
          ))}
        </Box>
      )}
      {chosenAircraft && (
        <AircraftCard
          data={chosenAircraft}
          onClick={clearChosenAircraft}
          chosen
        />
      )}
    </Box>
  );
};

export default BaseAircraftForm;
