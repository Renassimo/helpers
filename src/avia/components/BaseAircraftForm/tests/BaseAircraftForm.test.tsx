import userEvent from '@testing-library/user-event';
import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import useStateValue from '@/common/hooks/useStateValue';

import AviaInput from '@/avia/components/AviaInput';
import AircraftCard from '@/avia/components/AircraftCard';

import MockedAviaInput from '@/avia/components/AviaInput/mocks';
import MockedAircraftCard from '@/avia/components/AircraftCard/mocks';

import { mockedDeserializedAircrafts } from '@/avia/types/avia/mocks';

import BaseAircraftForm from '../BaseAircraftForm';
import { Avia } from '@/avia/types/avia';

jest.mock('@/common/hooks/useStateValue');
jest.mock('@/avia/components/AviaInput');
jest.mock('@/avia/components/AircraftCard');

describe('BaseAircraftForm', () => {
  const mockedSetValue = jest.fn();
  const mockUseStateValue = (value = '') =>
    jest.fn(() => [value, mockedSetValue]);

  const retreiveAircrafts = jest.fn();
  const chooseAircraft = jest.fn();
  const clearChosenAircraft = jest.fn();

  const aircraftsResult = {
    aircrafts: mockedDeserializedAircrafts,
    chosenAircraft: mockedDeserializedAircrafts[0],
    retreiveAircrafts,
    chooseAircraft,
    clearChosenAircraft,
    loading: false,
  } as unknown as Avia.AircraftsResult;

  const getAircraftsResult = (props: Partial<Avia.AircraftsResult> = {}) => ({
    ...aircraftsResult,
    ...props,
  });

  beforeEach(() => {
    (AviaInput as unknown as jest.Mock).mockImplementation(MockedAviaInput);
    (AircraftCard as unknown as jest.Mock).mockImplementation(
      MockedAircraftCard
    );
    (useStateValue as unknown as jest.Mock).mockImplementation(
      mockUseStateValue()
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(
      <BaseAircraftForm aircraftsResult={getAircraftsResult()} />
    );
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  describe('when registration passes', () => {
    test('calls mockedSetValue with registration', async () => {
      // Arange
      // Act
      renderWithTheme(
        <BaseAircraftForm
          registration="Registration"
          aircraftsResult={getAircraftsResult()}
        />
      );
      // Assert
      expect(mockedSetValue).toBeCalledWith('Registration');
    });
  });

  describe('when no chosenAircraft', () => {
    const contextProps = { chosenAircraft: null };

    test('renders successfully', () => {
      // Arange
      // Act
      const { baseElement } = renderWithTheme(
        <BaseAircraftForm aircraftsResult={getAircraftsResult(contextProps)} />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
    });

    describe('and when value is not empty', () => {
      const value = 'Val';

      test('renders successfully', async () => {
        // Arange
        (useStateValue as unknown as jest.Mock).mockImplementation(
          mockUseStateValue(value)
        );
        // Act
        const { baseElement } = renderWithTheme(
          <BaseAircraftForm
            aircraftsResult={getAircraftsResult(contextProps)}
          />
        );
        // Assert
        expect(baseElement).toMatchSnapshot();
      });

      describe('and clicks to search', () => {
        test('calls retreiveAircrafts', async () => {
          // Arange
          (useStateValue as unknown as jest.Mock).mockImplementation(
            mockUseStateValue(value)
          );
          const { getByLabelText } = renderWithTheme(
            <BaseAircraftForm
              aircraftsResult={getAircraftsResult(contextProps)}
            />
          );
          // Act
          await userEvent.click(getByLabelText('Search'));
          // Assert
          expect(retreiveAircrafts).toBeCalledWith('Val', false);
        });

        describe('and useOwnDb is true', () => {
          test('calls retreiveAircrafts', async () => {
            // Arange
            (useStateValue as unknown as jest.Mock).mockImplementation(
              mockUseStateValue(value)
            );
            const { getByLabelText } = renderWithTheme(
              <BaseAircraftForm
                aircraftsResult={getAircraftsResult(contextProps)}
                useOwnDb
              />
            );
            // Act
            await userEvent.click(getByLabelText('Search'));
            // Assert
            expect(retreiveAircrafts).toBeCalledWith('Val', true);
          });
        });

        describe('and uwhen checks "From own database"', () => {
          test('calls retreiveAircrafts', async () => {
            // Arange
            (useStateValue as unknown as jest.Mock).mockImplementation(
              mockUseStateValue(value)
            );
            const { getByLabelText } = renderWithTheme(
              <BaseAircraftForm
                aircraftsResult={getAircraftsResult(contextProps)}
              />
            );
            await userEvent.click(getByLabelText('From own database'));
            // Act
            await userEvent.click(getByLabelText('Search'));
            // Assert
            expect(retreiveAircrafts).toBeCalledWith('Val', true);
          });
        });
      });
    });
  });

  describe('when no aircrafts', () => {
    test('renders successfully', () => {
      // Arange
      // Act
      const { baseElement } = renderWithTheme(
        <BaseAircraftForm
          aircraftsResult={getAircraftsResult({
            chosenAircraft: null,
            aircrafts: null,
          })}
        />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
  });
});
