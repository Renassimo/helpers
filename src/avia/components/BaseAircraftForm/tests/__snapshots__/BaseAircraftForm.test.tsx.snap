// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseAircraftForm renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        MockedAircraftCard - 
        chosen
      </div>
      <label
        class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          aria-disabled="true"
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          tabindex="-1"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            disabled=""
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label Mui-disabled css-ahj2mt-MuiTypography-root"
        >
          From own database
        </span>
      </label>
    </div>
  </div>
</body>
`;

exports[`BaseAircraftForm when no aircrafts renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-178yklu"
      >
        <div>
          MockedAviaInput - 
          Aircraft registration
           - 
           - 
           - 
           -
           
           - 
          <button
            aria-label="Search"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </button>
        </div>
      </div>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          From own database
        </span>
      </label>
    </div>
  </div>
</body>
`;

exports[`BaseAircraftForm when no chosenAircraft and when value is not empty renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-178yklu"
      >
        <div>
          MockedAviaInput - 
          Aircraft registration
           - 
           - 
           - 
          Val
           -
           
           - 
          <button
            aria-label="Search"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-ohyvb6"
      >
        <div
          class="MuiBox-root css-hnroeo"
        >
          <div>
            MockedAircraftCard - 
          </div>
        </div>
        <div
          class="MuiBox-root css-12z0wuy"
        >
          <div>
            MockedAircraftCard - 
          </div>
        </div>
      </div>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          From own database
        </span>
      </label>
    </div>
  </div>
</body>
`;

exports[`BaseAircraftForm when no chosenAircraft renders successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-178yklu"
      >
        <div>
          MockedAviaInput - 
          Aircraft registration
           - 
           - 
           - 
           -
           
           - 
          <button
            aria-label="Search"
            class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="MuiBox-root css-ohyvb6"
      >
        <div
          class="MuiBox-root css-hnroeo"
        >
          <div>
            MockedAircraftCard - 
          </div>
        </div>
        <div
          class="MuiBox-root css-12z0wuy"
        >
          <div>
            MockedAircraftCard - 
          </div>
        </div>
      </div>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          From own database
        </span>
      </label>
    </div>
  </div>
</body>
`;
